Library/Formula/carina.rb

```diff
@@ -0,0 +1,30 @@
+class Carina < Formula
+  desc "Work with Swarm clusters on Carina"
+  homepage "https://github.com/getcarina/carina"
+  url "https://github.com/getcarina/carina.git",
+        :tag => "v0.7.1",
+        :revision => "885c4e298a9064e60a07dffd0143c38095def96e"
+  head "https://github.com/getcarina/carina.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    carinapath = buildpath/"src/github.com/getcarina/carina"
+    carinapath.install Dir["{*,.git}"]
+
+    cd carinapath do
+      version = `git describe --tags 2> /dev/null`.strip
+      commit = `git rev-parse --verify HEAD`.strip
+
+      system "make", "get-deps"
+      system "make", "carina", "COMMIT=#{commit}", "VERSION=#{version}"
+      bin.install "carina"
+    end
+  end
+
+  test do
+    system "#{bin}/carina", "--version"
```

Comment:
Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!